src/router.php

```diff
@@ -1,32 +1,23 @@
 <?php
 namespace Ericabwalker\PHPfinal;
 
-class Router 
-{
-    public $uri;
-    public $action;
-    public $method;
-    public $routes;
-
-    function get($uri, $action) {
-
-        // return 
-        // echo $uri . " " . $action;
+use Ericabwalker\PHPfinal\Controllers\BooksController;
 
+class Router
+{
+    public function get($uri, $action)
+    {
     }
 
-    function execute($routes) {
-        foreach($routes as $method => $route) {
-            if($method == $_SERVER['REQUEST_METHOD']) {
-                var_dump($route);
+    public function execute($routes)
+    {
+        $method = $_SERVER['REQUEST_METHOD'];
+        foreach ($routes[$method] as $uri => $action) {
+            if ($uri == $_SERVER['REQUEST_URI']) {
+                $method = explode('@', $action)[1];
+                $controller = new BooksController();
+                $controller->$method();
```

Comment:
You remembered that `explode` method is like a string split function in PHP, so we are using the first index, or right section, of `BooksController@add_books_form` as the `$method`. Then we call that method name using the `$controller->$method();` notation just like we saw in the `Request` class from other projects.